user-impl/src/main/java/com/example/auction/user/impl/UserRepository.java

```diff
@@ -0,0 +1,180 @@
+            private final CassandraReadSide readSide;
+
+            private PreparedStatement insertUserStatement;
+
+            @Inject
+            public PUserEventProcessor(CassandraSession session, CassandraReadSide readSide) {
+                this.session = session;
+                this.readSide = readSide;
+            }
+
+            @Override
+            public ReadSideHandler<PUserEvent> buildHandler() {
+                return readSide.<PUserEvent>builder("pUserEventOffset")
+                        .setGlobalPrepare(this::createTables)
+                        .setPrepare(tag -> prepareStatements())
+                        .setEventHandler(PUserEvent.PUserCreated.class,
+                                e -> insertUser(e.getUser()))
+
+                        .build();
+            }
+
+            @Override
+            public PSequence<AggregateEventTag<PUserEvent>> aggregateTags() {
+                return PUserEvent.TAG.allTags();
+            }
+
+            private CompletionStage<Done> createTables() {
+                return doAll(
+                        session.executeCreateTable(
+                                "CREATE TABLE IF NOT EXISTS UserInfo (" +
+                                        "UserId UUID PRIMARY KEY, " +
+                                        "Name text, "+
+                                        "email text, " +
+                                        "PasswordHash text "+
+                                        ")"
+                        )
+
+                );
+            }
+
+            private CompletionStage<Done> prepareStatements() {
+                return
+                        prepareInsertUserStatement();
+
+
+            }
+
+
+
+            private CompletionStage<Done> prepareInsertUserStatement() {
+
+                return session.
+                        prepare("INSERT INTO User(" +
+
+                                "UserId, " +
+                                "Name, " +
+                                "email, " +
+                                "PasswordHash " +
+
+                                ") VALUES (" +
+                                "?, " + // UserId
+                                "?, " + // Name
+                                "?, " + // email
+                                "?"   + // PasswordHash
+
+                                ")"
+                        )
+                        .thenApply(accept(s -> insertUserStatement = s ));
+            }
+
+
+
+
+            private CompletionStage<List<BoundStatement>> insertUser(PUser user) {
+                return completedStatements(
+                        (List<BoundStatement>) insertUserCreator(user)
+                );
+            }
+            private BoundStatement insertUserCreator(PUser user) {
+                return insertUserStatement.bind(
+
+                        user.getId(),
+                        user.getEmail(),
+                        user.getName()
```

Comment:
This doesn't look quite right. The statement you are binding to is this:

```
"INSERT INTO User(UserId, Name, email, PasswordHash) VALUES (?, ?, ?, ?)"
```

You need to bind the same number of parameters, in the same order they appear in the statement.